### PR TITLE
Implementation of most GoF patterns

### PR DESCRIPTION
**Pull request by philips-ts**

### CODE REVIEW

File: src/main/java/com/tarasenko/gof_patterns/behavioral/CommandPattern.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public void buttonPress() {
+        if (isStarted) {
+            stopEngineCommand.execute();
+            isStarted = false;
+        } else {
+            startEngineCommand.execute();
+            isStarted = true;
+        }
```
Comment (ALnOUA) [inline src/main/java/com/tarasenko/gof_patterns/behavioral/CommandPattern.java:42]: We can use this code without "else"

File: src/main/java/com/tarasenko/gof_patterns/creational/abstract_factory/GUIFactory.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    static GUIFactory of(String osName) {
+        osName = osName.toLowerCase();
+
+        if (osName.contains("mac")) {
+            return new MacOsFactory();
+        } else if (osName.contains("windows")){
+            return new WindowsFactory();
+        } else {
+            throw new UnsupportedOperationException("Unsupported OS version");
+        }
```
Comment (ALnOUA) [inline src/main/java/com/tarasenko/gof_patterns/creational/abstract_factory/GUIFactory.java]: The same

**Review by ALnOUA — CHANGES_REQUESTED** [review]
Try to optimize your code and not to use checking that are useless

File: src/main/java/com/tarasenko/gof_patterns/behavioral/CommandPattern.java (hunk 1 above)
Comment (philips-ts) [reply src/main/java/com/tarasenko/gof_patterns/behavioral/CommandPattern.java:42]: Left as is

File: src/main/java/com/tarasenko/gof_patterns/creational/abstract_factory/GUIFactory.java (hunk 2 above)
Comment (philips-ts) [reply src/main/java/com/tarasenko/gof_patterns/creational/abstract_factory/GUIFactory.java]: Done

**Review by ALnOUA — APPROVED** [review]

**Pull request merged**